nixos/modules/services/misc/mautrix-whatsapp.nix

```diff
@@ -0,0 +1,48 @@
+{ lib, config, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.mautrix-whatsapp;
+in
+{
+  options.services.mautrix-whatsapp = {
+    enable = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable mautrix-whatsapp, a puppeting bridge between Matrix and WhatsApp.
+
+        See <link xlink:href="https://github.com/tulir/mautrix-whatsapp/wiki/Bridge-setup">here</link> for documentation.
+     '';
+    };
+
+    config = mkOption {
+      type = types.lines;
+      description = "YAML configuration file for the bridge";
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.mautrix-whatsapp = {
+      description = "Mautrix-WhatsApp Service - A WhatsApp bridge for Matrix";
+      after = [ "network.target" ];
+      wantedBy = [ "multi-user.target" ];
+
+      serviceConfig = {
+        Type = "simple";
+        User = "mautrix-whatsapp";
+        WorkingDirectory = /var/lib/mautrix-whatsapp;
+        ExecStart = "${pkg.mautrix-whatsapp}/bin/mautrix-whatsapp -c ${pkgs.writeText "mwa-config.yaml" cfg.config}";
+        Restart = "on-failure";
+      };
+
+      users.users.mautrix-whatsapp = {
+        uid = config.ids.uids.mautrix-whatsapp;
```

Comment:
I guess this is not yet defined, and can go away if the service uses `DynamicUser`.

Reply:
I already heard about DynamicUser but how can I implement this ? Some docs somewhere ? (I didn't found yet)

Reply:
A fairly detailed explanation can be found in Lennart's blog: http://0pointer.net/blog/dynamic-users-with-systemd.html

Also, when you grep over `nixpkgs`, you'll probably find several modules that use `DynamicUser` already, so you have some real-world examples as well :)